### PR TITLE
Enable users to upload zip files to continue editing

### PR DESCRIPTION
**Pull request by Aritra8438**
# Description

This PR enables users to upload zip file so that they can modify it further.

https://github.com/user-attachments/assets/054c1cf1-eacb-4261-8dfe-164b8ca982fe

# Comments

- Users must upload the zip file they generated from the new CDP. It's not compatible with the unofficial CDP.
- The `cdp.data` file should be in the zip to upload for further modification.

# Checklist:

- [x] The code follows the [coding guidelines](https://github.com/omegaup/omegaup/wiki/Coding-guidelines) of omegaUp.
- [x] The tests were executed and all of them passed.
- [x] If you are creating a feature, the new tests were added.
- [ ] If the change is large (> 200 lines), this PR was split into various Pull Requests. It's preferred to create one PR for changes in controllers + unit tests in PHPUnit,  and then another Pull Request for UI + tests in Jest, Cypress or both.

### CODE REVIEW

**Comment by Aritra8438** [comment]
@carlosabcs PTAL!

File: frontend/www/js/omegaup/components/problem/creator/Header.vue [hunk 1]
```diff
@@ -34,11 +60,70 @@
...
+            if (storeData.casesStore) {
+              this.$store.commit(
+                'casesStore/replaceState',
+                storeData.casesStore,
+              );
+            }
+          });
+        } else {
+          ui.error(T.problemCreatorZipFileIsNotComplete);
+        }
```
Comment (carlosabcs) [inline frontend/www/js/omegaup/components/problem/creator/Header.vue:121]: Please, use the guard-clause pattern. You can start with:
```
if (!cdpDataFile) {
  ui.error(T.problemCreatorZipFileIsNotComplete);
}
```

**Comment by Aritra8438** [comment]
@carlosabcs, PTAL!

**Comment by carlosabcs** [comment]
@Aritra8438 please solve the conflicts and I'll do another review.

**Comment by Aritra8438** [comment]
@carlosabcs PTAL!

File: frontend/templates/en.lang [hunk 2]
```diff
@@ -1110,6 +1110,11 @@ problemCreatorTestCases = "Test cases"
 problemCreatorUngrouped = "Ungrouped"
 problemCreatorUngroupedCases = "Ungrouped cases"
 problemCreatorUpdateAlert = "The problem has been updated succesfully."
+problemCreatorUploadZip = "Upload"
+problemCreatorUploadZipMessage = "Please upload the zip file for the problem"
```
Comment (heduenas) [inline frontend/templates/en.lang:1114]: This specific message is phrased too formally for omegaUp. Can you change it to:
* EN: Upload the zip file of your problem here.
* ES: Sube el archivo zip de tu problema aquí.
* PT: Envie aqui o arquivo zip do seu problema.

@carlosabcs is the Portuguese translation correct?

Comment (Aritra8438) [reply frontend/templates/en.lang]: Done!

Comment (carlosabcs) [reply frontend/templates/en.lang]: It is

**Review by carlosabcs — APPROVED** [review]
LGTM

**Comment by Aritra8438** [comment]
@heduenas PTAL!

**Pull request merged**